### ISSUE TITLE
Simplify `ImageReaderSurfaceProducer`: Do not trim memory for external textures

### ISSUE THREAD

**Issue by matanlurey** [flutter/flutter#156488]
As of the migration from `SurfaceTexture` to `ImageReader`, we inadvertently broke the ability to run videos in the background, because when the application is backgrounded it typically receives a `TRIM_MEMORY(BACKGROUND)` event, which in turn causes the surface to be released.

For example, [this option is no longer functional](https://pub.dev/documentation/video_player/latest/video_player/VideoPlayerOptions/allowBackgroundPlayback.html):

```dart
VideoPlayerOptions({
  bool allowBackgroundPlayback = false,
})
```

@johnmccutchan mentioned that the particular workaround was for Android 14, and it's possible that in other versions (i.e. Android 15 or newer), we could support background playback (by not releasing the texture). However, I want to make sure (a) I understand this correctly, (b) we consider this a problem, and (c) have a plan on how to fix this - for now it's likely I need to just make a note this functionality is unsupported on Android (or folks will need to fallback to an older pre-ImageReader version of the plugin, which is not ideal).

**Comment by jonahwilliams** [comment]
Questions:

1. Do we know for sure that Android 15 will have fixed this
2. Do we have the ability to write an integration test that would break if this was re-broken?

i'd assume most Android 14 devices will update at some point to 15.

**Comment by matanlurey** [comment]
Let me chat with @johnmccutchan and come back here.

**Comment by jonahwilliams** [comment]
Adding some more details:

This bug is particularly problematic because it impacts both platform views and external textures. For the latter, we could make surface produer use GLES textures on Android 14. For the former, we don't have a mechanism to make that work.

If we're able to make HC platform views work better and succeed in switching most plugins (even video player) to using these surfaces then maybe this won't matter as much.

**Comment by matanlurey** [comment]
Ok, so our plan is as follows:

1. Accept that `video_player_android` does not support background video playback at the moment
2. After this stable release:
  - patch `SurfaceProducer` to use `SurfaceTexture` on Android 14
  - patch `SurfaceProducer` to no longer release textures aggressively

There might be some edge-cases around platform views v external textures to consider.

**Comment by gmstyle** [comment]
Please make sure to support the background video playback feature as before, because it's important. Thanks

**Comment by a-v-ebrahimi** [comment]
The app stops playback when I turn off screen, it's for same reason?

**Comment by matanlurey** [comment]
Yes, video playback while the app is not active (turned off, other app) is currently not supported.

**Comment by matanlurey** [comment]
Not yet complete.

**Comment by emakar** [comment]
For some reason I sometimes don't receive `onSurfaceAvailable` once resumed
I've checked `FlutterActivity.onResume()` is called, but `FlutterRenderer`'s `DefaultLifecycleObserver.onResume` - isn't

This happens if quickly navigating to home screen and back to the app

Pixel 6a, Flutter 3.29.0

**Comment by emakar** [comment]
Well, yes, `ProcessLifecycleOwner` is to blame
Breakpoint stops at `handler!!.removeCallbacks(delayedPauseRunnable)` when surface is black on resume (ie `onSurfaceAvailable` isn't called)

```

    internal fun activityResumed() {
        resumedCounter++
        if (resumedCounter == 1) {
            if (pauseSent) {
                registry.handleLifecycleEvent(Lifecycle.Event.ON_RESUME)
                pauseSent = false
            } else {
                handler!!.removeCallbacks(delayedPauseRunnable)
            }
        }
    }
```
Whereas in normal resumes `registry.handleLifecycleEvent(Lifecycle.Event.ON_RESUME)` is called

**Comment by emakar** [comment]
tldr - it's a bug in FlutterRenderer (because of using ProcessLifecycleOwner together with onTrimMemory)

if we pause-resume within 700ms
- neither ProcessLifecycleOwner onPause / onResume are sent
- but onTrimMemory when we pause

The both behave within their docs, and they are out of sync with each other. Described issue would be resolved if using Activity's lifecycle instead
